### ISSUE TITLE
Power Cost Discussion

### ISSUE THREAD

**Issue by Ipsis** [Ipsis/Woot#85]
If you have any comments, suggestions, complaints regarding the power costs.
Please ensure that you are running 1.10.2-0.2.4-alpha or above, as that is the baseline that I'll be using.

**Comment by Rated01010010** [comment]
Posted another comment in my last issue. Will now post any new ideas here.

**Comment by Ipsis** [comment]
So I've had more feedback on the power balance of this mod, especially when we dont have high RF output power gen mods out yet and by that I mean things like Big Reactors.

So I'm going to try and balance the costs based around a base cost of 80 RF/tick, which is what a fully upgraded EnderIO stirling generator can produce. All the other default power costs in the mod will be calculated around this value and multiples of it.
eg. a Tier I factory will require 80 RF/tick to operate. A zombie will cost 80 RF/tick to spawn. An enderman will cost 160 RF/tick to spawn.

As before, more valuable mobs will be more expensive to spawn. 

The only exception to this will be the rate upgrade.
The mod will still calculate how much to spawn the number of mobs you require in 320 ticks.
If you apply the rate upgrade, then you have to have the power infrastructure to generate the same total amount of power, in a shorter time frame.

So hopefully the next release will be a bit more user friendly.

**Comment by ShadowDrakken** [comment]
Glad to see this... right now I think the formulas are all busted up. I'm currently using the default config, and the tooltip on the blocks before placing them claim a Tier I should be 20RF/t, however once built and assign to produce cows, the base RF/t for a T1 with no upgrades jumps to 216RF/t o_O with a T1 looting and T1 speed it's over 500RF/t!

None of those numbers make any sense when I look at the numbers in the config file.

Hoping that the math can either be made more transparent, or if the formula is broken that it can be fixed along with this rebalance.

**Comment by Ipsis** [comment]
There is a power page on the wiki that shows how power is calculated.

Note that the tooltip shows how much RF/tick to run the factory tier. That does not include the cost to actually spawn the mobs, which includes the upgrades and how fancy the mob is.

I'm looking at an in-game manual to try and make it a bit clearer without using the wiki.

**Comment by Ipsis** [comment]
Also make sure you delete the config file to ensure that the correct values are being used. Cows should be cheapish. Of course I'm not ruling out bugs :)

Advanced generators I hadn't tried yet .... time to add it to my test instance.

**Comment by ShadowDrakken** [comment]
It's already a brand new config, and it's a T1 farm eating 216RF/t for cows... that seems a bit rough for cattle :)

**Comment by Ipsis** [comment]
Something is wrong there. From a quick look at the values, it should be

80 base + 16 per tick for a single 1xp mob. I'll need to check that when I get home. :(

**Comment by Ipsis** [comment]
Which version are you running?
I've just confirmed 96 RF/t for cows on a plain T1 factory.

**Comment by ShadowDrakken** [comment]
running 0.2.7
default config, just regenerated it to make sure --> http://pastebin.com/Y5wLacDW

![image](https://cloud.githubusercontent.com/assets/638689/17906392/1053ac36-6945-11e6-80a7-2303577f4ceb.png)

This screenshot has a T1 looting and T1 speed attached, each of which should be 80FR/t according to their own tooltips, but as you can see, the factory is eating 512RF/t rather than the 256RF/t it should be eating.

When I was building the factory, before I placed those two blocks it was 216RF/t :\
However, since I rebooted the server to regenerate teh config, when I remove the upgrades, it's correctly showing 96RF/t, but jumps back up to 512RF/t when I put them back on. So there's still something funky going on.

**Comment by Ipsis** [comment]
Pictures are always so helpful :)

There is nothing wrong, it is the speed upgrade.

Basically you have to provide a certain RF/tick for 320 ticks to spawn 1 mob, therefore there is a total RF for the spawn. The RF/tick depends on the mob, the upgrades and the tier of factory.

However when you add the rate/speed upgrade  you still have to provide the same total RF, but in less time. This means that the RF/tick goes up. The cost for a fully upgraded Tier IV with speed/rate III and mass III for Wither Bosses is somewhat eyewatering.

I would try with the mass upgrade which increases the number of mobs generated in the 320 ticks, but it only costs you 33% extra for each mob.

The rate/speed upgrade is expensive as it teaches people patience ;)

**Comment by ShaCraft** [comment]
What do you think about having the Factory create Mob juice? You know Mob essence that could be compatible with mods like EnderIO. Maybe something to turn XP shards into goo? 

Another thought would it be very hard to have a block on the factory pull some form of image to show what mob is being farmed in the factory. Just some random thoughts I had setting up a factory in my series.

**Comment by Ipsis** [comment]
One of the reasons I added the proxy was to give some space to output fluid :)
So I had intended to output mob essence (MFR?) or equivalent or even liquid XP when it resurfaces.
The main issue would be trying to ensure it was exploitable.
So It is certainly on the cards, just not sure how I would do it.

As for displaying the mob, I want to do this and there is a bug somewhere about it. I just need to find some way of doing it that fits within the Forge "rules" for modding eg. not using a TESR. So hopefully I can find some way of implementing it, cause I think it would look good - well possibly - as well ;)
